File: sandbox-runner/patterns/event-accumulator/go/activities.go
package main

import (
	"context"
	"fmt"
	"strings"
)

// ProcessItems is the batch activity that processes all accumulated items together.
func ProcessItems(_ context.Context, orderID string, items []OrderItem) (string, error) {
	names := make([]string, len(items))
	for i, item := range items {
		names[i] = item.Name
	}
	return fmt.Sprintf("Order %s fulfilled with %d item(s): %s",
		orderID, len(items), strings.Join(names, ", ")), nil
}

File: sandbox-runner/patterns/event-accumulator/java/src/main/java/Activities.java
import io.temporal.activity.ActivityInterface;

import java.util.List;
import java.util.stream.Collectors;

@ActivityInterface
public interface Activities {
    String processItems(String orderId, List<Shared.OrderItem> items);

    class Impl implements Activities {
        @Override
        public String processItems(String orderId, List<Shared.OrderItem> items) {
            String names = items.stream()
                    .map(i -> i.name)
                    .collect(Collectors.joining(", "));
            return "Order " + orderId + " fulfilled with " + items.size() + " item(s): " + names;
        }
    }
}
